dashboard/app/models/user.rb

```diff
@@ -502,8 +511,8 @@ def user_progress_by_stage(stage)
   end
 
   def user_level_for(script_level, level)
-    user_levels.find_by(script_id: script_level.script_id,
-                        level_id: level.id)
+    user_levels_for_script = cached_user_levels_by_script(script_level.script_id)
+    user_levels_for_script[level.id]
```

Comment:
This method now fetches an entire script of `user_levels`, rather than one. This seems like a regression...